File: mobile/src/main/java/com/yalantis/watchface/task/SendToDataLayerThread.java
package com.yalantis.watchface.task;

import android.graphics.Bitmap;
import android.util.Log;

import com.google.android.gms.common.api.GoogleApiClient;
import com.google.android.gms.wearable.Node;
import com.google.android.gms.wearable.NodeApi;
import com.google.android.gms.wearable.Wearable;

import java.io.ByteArrayOutputStream;

public class SendToDataLayerThread extends Thread {

    private static final int MAX_SIZE = 200000000;
    private static final String MESSAGE_SUCESS=" was sent successfully";
    private static final String FAIL_SIZE="Big image file, try to use another";
    private String path;
    private Bitmap bitmap;
    private GoogleApiClient mGoogleApiClient;
    private DataLayerListener mDataLayerListener;

    public SendToDataLayerThread(String path, Bitmap bitmap, GoogleApiClient googleApiClient, DataLayerListener dataLayerListener) {
        this.path = path;
        this.bitmap = bitmap;
        mGoogleApiClient = googleApiClient;
        mDataLayerListener = dataLayerListener;
    }

    public void run() {
        NodeApi.GetConnectedNodesResult nodes = Wearable.NodeApi.getConnectedNodes(mGoogleApiClient).await();
        ByteArrayOutputStream stream = new ByteArrayOutputStream();
        bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
        String message = path + MESSAGE_SUCESS;
        if (bitmap.getByteCount() < MAX_SIZE) {
            for (Node node : nodes.getNodes()) {
                Wearable.MessageApi.sendMessage(mGoogleApiClient, node.getId(), path,  stream.toByteArray()).await();
            }
        } else {
            message = FAIL_SIZE;
        }
        mDataLayerListener.onSuccess(message);
    }

    public interface DataLayerListener  {

        void onSuccess(String message);

    }

}

File: mobile/src/main/java/com/yalantis/watchface/task/SendToDataLayerThread.kt
package com.yalantis.watchface.task

import android.graphics.Bitmap
import com.google.android.gms.common.api.GoogleApiClient
import com.google.android.gms.wearable.Wearable
import java.io.ByteArrayOutputStream

class SendToDataLayerThread(
        private val path: String,
        private val bitmap: Bitmap,
        private val mGoogleApiClient: GoogleApiClient,
        private val mDataLayerListener: DataLayerListener
) : Thread() {

    companion object {
        private const val MAX_SIZE = 2000000
        private const val MESSAGE_ERROR = "Big image file, try to use another"
        private const val MESSAGE_OK = "was sent successfully"
        private const val QUALITY_IMG = 100
    }

    override fun run() {
        val nodes = Wearable.NodeApi.getConnectedNodes(mGoogleApiClient).await()
        val stream = ByteArrayOutputStream()
        bitmap.compress(Bitmap.CompressFormat.PNG, QUALITY_IMG, stream)
        var message = "$path $MESSAGE_OK"
        if (bitmap.byteCount < MAX_SIZE) {
            for (node in nodes.nodes) {
                Wearable.MessageApi.sendMessage(mGoogleApiClient, node.id, path, stream.toByteArray()).await()
            }
        } else {
            message = MESSAGE_ERROR
        }
        mDataLayerListener.onSuccess(message)
    }

    interface DataLayerListener {
        fun onSuccess(message: String)
    }
}
